### PR TITLE
LPS-136247

### PR DESCRIPTION
**Pull request by DiegoDOC**
Related Issue: https://issues.liferay.com/browse/LPS-136247

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -12,239 +12,145 @@
...
+		const expectedDate = moment().format('MM/DD/YYYY');
 
-		await wait(() =>
-			expect(
-				getAllByDisplayValue(moment().format('MM/DD/YYYY'))
-			).toHaveLength(2)
-		);
-
-		expect(handleFieldEdited).toHaveBeenCalled();
+		expect(onChange).toHaveBeenCalled();
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:94]: According to test description this assertion is out of it scope 
```suggestion
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 2]
```diff
@@ -12,239 +12,145 @@
...
-		});
-
-		const date = moment().format('YYYY-MM-DD');
-
-		await wait(() => expect(getAllByDisplayValue(date)).toBeTruthy());
+		const displayDate = moment().format('MM/DD/YYYY');
+		const formattedDate = moment().format('YYYY-MM-DD');
 
-		expect(onChange).toHaveBeenCalledWith({}, date);
+		expect(getAllByDisplayValue(displayDate)).toBeTruthy();
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:115]: According to test description this assertion is out of it scope 
```suggestion
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 3]
```diff
@@ -12,239 +12,145 @@
...
 
-		await wait(() =>
-			expect(
-				document.body.querySelector('.date-picker-dropdown-menu.show')
-			).toBeTruthy()
+		const dropdownDatePicker = document.body.querySelector(
+			'.date-picker-dropdown-menu.show'
 		);
+
+		expect(dropdownDatePicker).toBeTruthy();
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:75]: Prefe use [jest-dom assertions](https://github.com/testing-library/jest-dom) for a proper error messages whenever the test fail:
```suggestion
		expect(dropdownDatePicker).tobeinthedocument();
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 4]
```diff
@@ -12,239 +12,145 @@
...
 
-		await wait(() =>
-			expect(
-				getAllByDisplayValue(moment().format('MM/DD/YYYY'))
-			).toHaveLength(2)
-		);
-
-		expect(handleFieldEdited).toHaveBeenCalled();
+		expect(onChange).toHaveBeenCalled();
+		expect(getAllByDisplayValue(expectedDate)).toHaveLength(2);
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:95]: - Try to query element by role or label to ensure the component's accessibility, according to the [query priority](https://testing-library.com/docs/queries/about/#priority) 
- Use [tohavevalue](https://github.com/testing-library/jest-dom#tohavevalue) to check the value of an `input` element:
```suggestion
		expect(getByRole('textbox')).tohavevalue(displayDate);
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 5]
```diff
@@ -12,239 +12,145 @@
...
+				predefinedValue="2020-06-02"
 			/>
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
+		const inputDatePicker = container
+			.getElementsByClassName('form-control')
+			.item(0);
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:57]: Try to query element by role or label to ensure the component's accessibility, according to the [query priority](https://testing-library.com/docs/queries/about/#priority) 
```suggestion
		const inputDatePicker = getByRole('textbox');
```

note that if many items are being returned you should try to make the query more specific and as last resource use `queryAllBy...`

**Review by miltonmc — CHANGES_REQUESTED** [review]
Great work!

Just some suggestions to improve the code:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 6]
```diff
@@ -14,89 +14,15 @@
...
-	const dateFormat = getLocaleDateFormat(locale);
-	const inputMask = getMaskByDateFormat(dateFormat);
-	const dateDelimiter = getDelimiter(inputMask);
-
-	return {
-		dateMask: getDateMask(dateDelimiter, dateFormat),
-		inputMask,
-	};
-};
+const NON_NUMERIC_REGEX = /[\D]/g;
```
[the remark this answers is not in the document]
Comment (DiegoDOC) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:25]: Thanks for the info!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js (hunk 5 above)
Comment (DiegoDOC) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:57]: Makes sense! Will be taking that in consideration next time.

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
> Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

@liferay-forms, I just ran the tests locally, and both of them seem to be failing on my master branch as well

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Comment by DiegoDOC** [comment]
ci:test:sf

**Comment by DiegoDOC** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @DiegoDOC, can you please check if these failures are related to your changes? Thanks

**Pull request closed without merging**